internal/pkg/metrics/controller_metrics.go

```diff
@@ -1,71 +1,127 @@
+	FeatureGatesEnabled *prometheus.GaugeVec
+	Paused              *prometheus.GaugeVec
+	ResourceSync        *prometheus.CounterVec
+}
+
 type ThanosQueryMetrics struct {
-	EndpointsConfigured                        *prometheus.GaugeVec
-	ServiceWatchesReconciliationsTotal         prometheus.Counter
-	FrontendServiceWatchesReconciliationsTotal prometheus.Counter
+	*CommonMetrics
+	EndpointsConfigured                *prometheus.GaugeVec
+	ServiceWatchesReconciliationsTotal *prometheus.CounterVec
 }
 
 type ThanosReceiveMetrics struct {
+	*CommonMetrics
 	HashringsConfigured                 *prometheus.GaugeVec
-	EndpointWatchesReconciliationsTotal prometheus.Counter
 	HashringHash                        *prometheus.GaugeVec
+	HashringTenantsConfigured           *prometheus.GaugeVec
+	HashringEndpointsConfigured         *prometheus.GaugeVec
+	EndpointWatchesReconciliationsTotal *prometheus.CounterVec
 }
 
 type ThanosRulerMetrics struct {
-	EndpointsConfigured                  *prometheus.GaugeVec
-	RuleFilesConfigured                  *prometheus.GaugeVec
-	ServiceWatchesReconciliationsTotal   prometheus.Counter
-	ConfigMapWatchesReconciliationsTotal prometheus.Counter
+	*CommonMetrics
+	EndpointsConfigured                       *prometheus.GaugeVec
+	RuleFilesConfigured                       *prometheus.GaugeVec
+	PrometheusRulesFound                      *prometheus.GaugeVec
+	PrometheusRuleGroupsFound                 *prometheus.GaugeVec
+	PrometheusRuleGroupsTenantCount           *prometheus.GaugeVec
+	ConfigMapsCreated                         *prometheus.CounterVec
+	ConfigMapCreationFailures                 *prometheus.CounterVec
+	ServiceWatchesReconciliationsTotal        *prometheus.CounterVec
+	ConfigMapWatchesReconciliationsTotal      *prometheus.CounterVec
+	PrometheusRuleWatchesReconciliationsTotal *prometheus.CounterVec
 }
 
 type ThanosStoreMetrics struct {
+	*CommonMetrics
+	ShardsConfigured            *prometheus.GaugeVec
+	ShardCreationUpdateFailures *prometheus.CounterVec
 }
 
 type ThanosCompactMetrics struct {
+	*CommonMetrics
+	ShardsConfigured            *prometheus.GaugeVec
+	ShardCreationUpdateFailures *prometheus.CounterVec
 }
 
-func NewThanosQueryMetrics(reg prometheus.Registerer) ThanosQueryMetrics {
+var (
+	commonMetricsInstance *CommonMetrics
+	commonMetricsOnce     sync.Once
+)
+
+func NewCommonMetrics(reg prometheus.Registerer) *CommonMetrics {
+	commonMetricsOnce.Do(func() {
+		commonMetricsInstance = &CommonMetrics{
+			FeatureGatesEnabled: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+				Name: "thanos_operator_feature_gates_enabled",
+				Help: "Feature gates enabled for ThanosOperator",
+			}, []string{"component"}),
+			Paused: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+				Name: "thanos_operator_paused",
+				Help: "Paused state of ThanosOperator",
+			}, []string{"component", "resource", "namespace"}),
+			ResourceSync: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
```

Comment:
I think this particular one, both success and failures as well as panics are caught in ctrl runtime metrics as per 
<img width="582" alt="image" src="https://github.com/user-attachments/assets/70aac693-228a-4d32-b26f-2afb1e412d1e" />